In [1]:
import geopandas as gpd
import pandas as pd
import numpy as np

In [2]:
# read in KNN-Access data analysis results 
df = gpd.read_file('../data/aziz-final/Seattle Output Files/origin_record.geojson')

In [4]:
df.head()

Baseline  UG1 Network  UG2 Network  UG3 Network  UG1 Node  UG2 Node  \
0  5.779995     2.214172     0.728205     0.000000  5.474603  0.455625   
1  5.779995     2.214172     0.728205     0.000000  5.474603  0.455625   
2  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   
3  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   
4  9.914821     9.753530     7.781496     5.035237  9.895874  9.858677   

   UG3 Node  UG1 Elevation  UG2 Elevation  UG3 Elevation  UG1 Network+Node  \
0  0.000000       5.761933       5.727985       5.608296          1.740703   
1  0.000000       5.761933       5.727985       5.608296          1.740703   
2  9.825177       9.851188       9.740542       9.728332          9.741421   
3  9.825177       9.851188       9.740542       9.728332          9.741421   
4  9.825177       9.851188       9.740542       9.728332          9.741421   

   UG2 Network+Node  UG3 Network+Node  UG1 Network+Node+Elevation  \
0          0.000000          0.000000                    1.387189   
1          0.000000          0.000000                    1.387189   
2          7.866494          5.037969                    9.686386   
3          7.866494          5.037969                    9.686386   
4          7.866494          5.037969                    9.686386   

   UG2 Network+Node+Elevation  UG3 Network+Node+Elevation  \
0                    0.000000                    0.000000   
1                    0.000000                    0.000000   
2                    7.400217                    4.966901   
3                    7.400217                    4.966901   
4                    7.400217                    4.966901   

                         geometry  
0  POINT (1260267.129 243780.167)  
1  POINT (1260267.129 243780.167)  
2  POINT (1260483.365 243274.057)  
3  POINT (1260483.365 243274.057)  
4  POINT (1260483.365 243274.057)

# Differences between Baseline and UG1, UG2, UG3

In [7]:
df_difference= df[['Baseline', 'UG1 Network+Node+Elevation','UG2 Network+Node+Elevation','UG3 Network+Node+Elevation']]

In [9]:
df_difference.head()

Baseline  UG1 Network+Node+Elevation  UG2 Network+Node+Elevation  \
0  5.779995                    1.387189                    0.000000   
1  5.779995                    1.387189                    0.000000   
2  9.914821                    9.686386                    7.400217   
3  9.914821                    9.686386                    7.400217   
4  9.914821                    9.686386                    7.400217   

   UG3 Network+Node+Elevation  
0                    0.000000  
1                    0.000000  
2                    4.966901  
3                    4.966901  
4                    4.966901

In [25]:
df_difference['UG1_diff'] =  df_difference['Baseline'] - df_difference['UG1 Network+Node+Elevation']
df_difference['UG2_diff'] =  df_difference['Baseline'] - df_difference['UG2 Network+Node+Elevation']
df_difference['UG3_diff'] =  df_difference['Baseline'] - df_difference['UG3 Network+Node+Elevation']

/var/folders/zm/x5dzdlrs5mlbxrm7z5g6p05r0000gn/T/ipykernel_99016/1405292741.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_difference['UG1_diff'] =  df_difference['Baseline'] - df_difference['UG1 Network+Node+Elevation']
/var/folders/zm/x5dzdlrs5mlbxrm7z5g6p05r0000gn/T/ipykernel_99016/1405292741.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_difference['UG2_diff'] =  df_difference['Baseline'] - df_difference['UG2 Network+Node+Elevation']
/var/folders/zm/x5dzdlrs5mlbxrm7z5g6p05r0000gn/T/ipyke

In [20]:
df_difference

Baseline  UG1 Network+Node+Elevation  UG2 Network+Node+Elevation  \
0      5.779995                    1.387189                    0.000000   
1      5.779995                    1.387189                    0.000000   
2      9.914821                    9.686386                    7.400217   
3      9.914821                    9.686386                    7.400217   
4      9.914821                    9.686386                    7.400217   
...         ...                         ...                         ...   
49037  7.909613                    7.307682                    6.179019   
49038  7.910426                    7.309605                    6.178522   
49039  7.909620                    7.307697                    6.179015   
49040  7.937756                    7.785505                    6.659394   
49041  7.922992                    7.778083                    6.648903   

       UG3 Network+Node+Elevation  UG1_diff  UG2_diff  UG3_diff  
0                        0.000000  4.392806  5.779995  5.779995  
1                        0.000000  4.392806  5.779995  5.779995  
2                        4.966901  0.228435  2.514604  2.514604  
3                        4.966901  0.228435  2.514604  2.514604  
4                        4.966901  0.228435  2.514604  2.514604  
...                           ...       ...       ...       ...  
49037                    4.642198  0.601931  1.730594  1.730594  
49038                    4.643814  0.600821  1.731904  1.731904  
49039                    4.642211  0.601923  1.730605  1.730605  
49040                    4.177272  0.152250  1.278361  1.278361  
49041                    4.165577  0.144909  1.274089  1.274089  

[49042 rows x 7 columns]

In [26]:
df_difference[df_difference['UG1_diff'] < 0].count()

Baseline                      5305
UG1 Network+Node+Elevation    5305
UG2 Network+Node+Elevation    5305
UG3 Network+Node+Elevation    5305
UG1_diff                      5305
UG2_diff                      5305
UG3_diff                      5305
dtype: int64

In [27]:
df_difference[df_difference['UG2_diff'] < 0].count()

Baseline                      1430
UG1 Network+Node+Elevation    1430
UG2 Network+Node+Elevation    1430
UG3 Network+Node+Elevation    1430
UG1_diff                      1430
UG2_diff                      1430
UG3_diff                      1430
dtype: int64

In [28]:
df_difference[df_difference['UG3_diff'] < 0].count()

Baseline                      297
UG1 Network+Node+Elevation    297
UG2 Network+Node+Elevation    297
UG3 Network+Node+Elevation    297
UG1_diff                      297
UG2_diff                      297
UG3_diff                      297
dtype: int64

In [31]:
5305+1430+ 297

7032

Conclusion: There are 7,032 (14.3%) points where the KNN-Access Score for Network+Node+Elevation is lower than the Baseline, which are incorrect.

# Fixable Baseline

Differences between UG1 Elevation and UG1 Network+Node+Elevation, UG2 Elevation and UG2 Network+Node+Elevation, UG3 Elevation and UG3 Network+Node+Elevation

In [33]:
df_gap= df[['UG1 Elevation', 'UG2 Elevation','UG3 Elevation','UG1 Network+Node+Elevation','UG2 Network+Node+Elevation','UG3 Network+Node+Elevation']]

In [35]:
df_gap.head()

UG1 Elevation  UG2 Elevation  UG3 Elevation  UG1 Network+Node+Elevation  \
0       5.761933       5.727985       5.608296                    1.387189   
1       5.761933       5.727985       5.608296                    1.387189   
2       9.851188       9.740542       9.728332                    9.686386   
3       9.851188       9.740542       9.728332                    9.686386   
4       9.851188       9.740542       9.728332                    9.686386   

   UG2 Network+Node+Elevation  UG3 Network+Node+Elevation  
0                    0.000000                    0.000000  
1                    0.000000                    0.000000  
2                    7.400217                    4.966901  
3                    7.400217                    4.966901  
4                    7.400217                    4.966901

In [36]:
df_gap['UG1_gap'] =  df_gap['UG1 Elevation'] - df_gap['UG1 Network+Node+Elevation']
df_gap['UG2_gap'] =  df_gap['UG2 Elevation'] - df_gap['UG2 Network+Node+Elevation']
df_gap['UG3_gap'] =  df_gap['UG3 Elevation'] - df_gap['UG3 Network+Node+Elevation']

/var/folders/zm/x5dzdlrs5mlbxrm7z5g6p05r0000gn/T/ipykernel_99016/691118760.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_gap['UG1_gap'] =  df_gap['UG1 Elevation'] - df_gap['UG1 Network+Node+Elevation']
/var/folders/zm/x5dzdlrs5mlbxrm7z5g6p05r0000gn/T/ipykernel_99016/691118760.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_gap['UG2_gap'] =  df_gap['UG2 Elevation'] - df_gap['UG2 Network+Node+Elevation']
/var/folders/zm/x5dzdlrs5mlbxrm7z5g6p05r0000gn/T/ipykernel_99016/691118760.py:3: SettingWi

In [37]:
df_gap[df_gap['UG1_gap'] < 0].count()

UG1 Elevation                 7977
UG2 Elevation                 7977
UG3 Elevation                 7977
UG1 Network+Node+Elevation    7977
UG2 Network+Node+Elevation    7977
UG3 Network+Node+Elevation    7977
UG1_gap                       7977
UG2_gap                       7977
UG3_gap                       7977
dtype: int64

In [38]:
df_gap[df_gap['UG2_gap'] < 0].count()

UG1 Elevation                 1927
UG2 Elevation                 1927
UG3 Elevation                 1927
UG1 Network+Node+Elevation    1927
UG2 Network+Node+Elevation    1927
UG3 Network+Node+Elevation    1927
UG1_gap                       1927
UG2_gap                       1927
UG3_gap                       1927
dtype: int64

In [39]:
df_gap[df_gap['UG3_gap'] < 0].count()

UG1 Elevation                 570
UG2 Elevation                 570
UG3 Elevation                 570
UG1 Network+Node+Elevation    570
UG2 Network+Node+Elevation    570
UG3 Network+Node+Elevation    570
UG1_gap                       570
UG2_gap                       570
UG3_gap                       570
dtype: int64

Conclusion: There are 10,474 (21.4%) points where the KNN-Access Score for Network+Node+Elevation is lower than the Elevation (Fixable Baseline), which are incorrect.
